docs/securing-control-plane.md

```diff
@@ -0,0 +1,154 @@
+antrea-agent, kube-apiserver) to trust itself.
+
+Typically, clients that wish to access the antrea-controller API can
+authenticate server by validating against the CA certificate published in the
+`kube-system/antrea-ca` ConfigMap.
+
+## Providing your own certificates
+
+Since Antrea v0.7.0, you can provide your own certificates to Antrea. To do so,
+you must set the `selfSignedCert` field of `antrea-controller.conf` to `false`,
+so that the antrea-controller will read the certificate key pair from the
+`kube-system/antrea-controller-tls` Secret.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    app: antrea
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    selfSignedCert: false
+```
+
+You can generate the required certificates manually, or through
+[cert-manager](https://cert-manager.io/docs/). In either way, the certificate
+must be issued with the following key usages and DNS names:
+
+**Note: It assumes you are using `cluster.local` as the cluster domain, you
+should replace it with the actual one of your Kubernetes cluster.**
+
+X509 key usages:
+- digital signature
+- key encipherment
+- server auth
+
+DNS names:
+- antrea.kube-system.svc
+- antrea.kube-system.svc.cluster.local
+
+You can then create the `kube-system/antrea-controller-tls` Secret with the
+certificate key pair and the CA certificate in the following form:
+```yaml
+apiVersion: v1
+kind: Secret
+# The type can also be Opaque.
+type: kubernetes.io/tls
+metadata:
+  name: antrea-controller-tls
+  namespace: kube-system
+data:
+  ca.crt: <BASE64 ENCODED CA CERTIFICATE>
+  tls.crt: <BASE64 ENCODED TLS CERTIFICATE>
+  tls.key: <BASE64 ENCODED TLS KEY>
+```
+
+### Using kubectl
+
+You can use `kubectl apply -f <PATH TO SECRET YAML>` to create the above secret,
+or use `kubectl create secret`:
+```bash
+kubectl create secret generic antrea-controller-tls -n kube-system \
+  --from-file=ca.crt=<PATH TO CA CERTIFICATE> --from-file=tls.crt=<PATH TO TLS CERTIFICATE> --from-file=tls.key=<PATH TO TLS KEY> 
+```
+
+### Using cert-manager
+
+If you set up [cert-manager](https://cert-manager.io/docs/) to manage your
+certificates, it can be used to issue and renew the certificate required by
+Antrea.
+
+To get started, follow the [cert-manager installation documentation](
+https://cert-manager.io/docs/installation/kubernetes/) to deploy cert-manager
+and configure `Issuer` or `ClusterIssuer` resources.
+
+The `Certificate` should be created in the `kube-system` namespace. For example,
+A `Certificate` may look like:
+```yaml
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: antrea-controller-tls
+  namespace: kube-system
+spec:
+  secretName: antrea-controller-tls
+  commonName: antrea
+  dnsNames:
+  - antrea.kube-system.svc
+  - antrea.kube-system.svc.cluster.local
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+  issuerRef:
+    name: ca-issuer
```

Comment:
the name can vary here, correct? maybe add a comment

Reply:
done